Submitted By: Bruce dubbs (bdubbs at linuxfromscratch dot org)
Date: 2010-01-28
Initial Package Version: 5.1.42
Origin: mysql mailing list
Description: Patches the mysqlclient library to fix a problem when
             compiled with gcc-4.4.

--- a/libmysql/libmysql.c  2009-11-03 18:18:44 +0000
+++ b/libmysql/libmysql.c  2009-11-23 21:54:27 +0000
@@ -2285,7 +2285,7 @@ mysql_stmt_param_metadata(MYSQL_STMT *st

 /* Store type of parameter in network buffer. */

-static void store_param_type(char **pos, MYSQL_BIND *param)
+static void store_param_type(unsigned char **pos, MYSQL_BIND *param)
 {
   uint typecode= param->buffer_type | (param->is_unsigned ? 32768 : 0);
   int2store(*pos, typecode);
@@ -2565,7 +2565,7 @@ int cli_stmt_execute(MYSQL_STMT *stmt)
   that is sent to the server.
       */
       for (param= stmt->params; param < param_end ; param++)
-        store_param_type((char**) &net->write_pos, param);
+        store_param_type(&net->write_pos, param);
     }

     for (param= stmt->params; param < param_end; param++)

